lib/Doctrine/DBAL/Schema/OracleSchemaManager.php

```diff
@@ -36,6 +36,52 @@ class OracleSchemaManager extends AbstractSchemaManager
     /**
      * {@inheritdoc}
      */
+    public function listTables()
+    {
+        $currentDatabase = $this->_conn->getDatabase();
+
+        $tableNames = $this->listTableNames();
+
+        // Get all column definitions in one database call.
+        $allColumns = $this->_conn->fetchAll($this->_platform->getListAllColumnsSQL($currentDatabase));
+
+        // Get all foreign keys definitions in one database call.
+        $allForeignKeys = $this->_conn->fetchAll($this->_platform->getListAllForeignKeysSQL($currentDatabase));
+
+        // Get all indexes definitions in one database call.
+        $allIndexes = $this->_conn->fetchAll($this->_platform->getListAllIndexesSQL($currentDatabase));
+
+        $tables = [];
+        foreach ($tableNames as $tableName) {
+            $unquotedTableName = rtrim(ltrim($tableName, '"'), '"');
+
+            // Process columns for this table.
+            $tableColumns = array_filter($allColumns, function ($column) use ($unquotedTableName) {
+                return $column['TABLE_NAME'] === $unquotedTableName;
+            });
+            $columns = $this->_getPortableTableColumnList($tableName, null, $tableColumns);
+
+            // Process foreign keys for this table.
+            $tableForeignKeys = array_filter($allForeignKeys, function ($foreignKey) use ($unquotedTableName) {
+                return $foreignKey['TABLE_NAME'] === $unquotedTableName;
+            });
+            $foreignKeys = !empty($tableForeignKeys) ? $this->_getPortableTableForeignKeysList($tableForeignKeys) : [];
```

Comment:
Not really worth optimization IMO. Calling `_getPortableTableForeignKeysList()` with an empty array will produce an empty array which is fine. Especially if called once before grouping by table name.